packages/build-tools/src/steps/functions/repack.ts

```diff
@@ -0,0 +1,187 @@
+        required: true,
+      }),
+      BuildStepInput.createProvider({
+        id: 'output_path',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: false,
+      }),
+      BuildStepInput.createProvider({
+        id: 'working_directory',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: false,
+      }),
+      BuildStepInput.createProvider({
+        id: 'embed_bundle_assets',
+        allowedValueTypeName: BuildStepInputValueTypeName.BOOLEAN,
+        required: false,
+      }),
+    ],
+    outputProviders: [
+      BuildStepOutput.createProvider({
+        id: 'output_path',
+        required: true,
+      }),
+    ],
+    fn: async (stepsCtx, { inputs, outputs, env }) => {
+      const projectRoot = stepsCtx.workingDirectory;
+      const verbose = stepsCtx.global.env['EAS_VERBOSE'] === '1';
+      const repackLogger = createBunyanLoggerAdapter(stepsCtx.logger);
+      const repackSpawnAsync = createSpawnAsyncStepAdapter(verbose, stepsCtx.logger);
+
+      const tmpDir = path.join(os.tmpdir(), `repack-${uuidv4()}`);
+      const workingDirectory =
+        (inputs.working_directory.value as string) ?? path.join(tmpDir, 'working-directory');
+      await fs.promises.mkdir(workingDirectory, { recursive: true });
+      stepsCtx.logger.info(`Created temporary directory: ${tmpDir}`);
+
+      const sourceAppPath = inputs.source_app.value as string;
+      const outputPath =
+        (inputs.output_path.value as string) ??
+        createDefaultOutputPath({ tmpDir, job: stepsCtx.global.staticContext.job });
+
+      stepsCtx.logger.info('Repacking the app...');
+      if (stepsCtx.global.staticContext.job.platform === Platform.IOS) {
+        await repackAppIosAsync({
+          platform: 'ios',
+          projectRoot,
+          sourceAppPath,
+          outputPath,
+          workingDirectory,
+          // TODO: add iosSigningOptions
+          logger: repackLogger,
+          spawnAsync: repackSpawnAsync,
+          verbose,
+          env: {
+            FASTLANE_DISABLE_COLORS: '1',
+            FASTLANE_SKIP_UPDATE_CHECK: '1',
+            SKIP_SLOW_FASTLANE_WARNING: 'true',
+            FASTLANE_HIDE_TIMESTAMP: 'true',
+            LC_ALL: 'en_US.UTF-8',
+            ...env,
+          },
+        });
+      } else if (stepsCtx.global.staticContext.job.platform === Platform.ANDROID) {
+        await repackAppAndroidAsync({
+          platform: 'android',
+          projectRoot: stepsCtx.workingDirectory,
+          sourceAppPath,
+          outputPath,
+          workingDirectory,
+          // TODO: add androidSigningOptions
+          logger: repackLogger,
+          spawnAsync: repackSpawnAsync,
+          verbose,
+          env,
+        });
+      } else {
+        throw new Error('Unsupported platform');
+      }
+
+      if (!(await pathExistsAsync(outputPath))) {
+        throw new Error(`Failed to repack the app. ${outputPath} does not exist`);
```

Comment:
This feels like a weird error message. Shouldn't `repackApp[Platform]Async` throw if it fails to repack the app?